src/common/FlagsEnum.hpp

```diff
@@ -65,43 +80,59 @@ class FlagsEnum
         }
     }
 
-    bool has(T flag) const
+    constexpr FlagsEnum operator|(T flag) const noexcept
     {
-        return static_cast<Q>(this->value_) & static_cast<Q>(flag);
+        return FlagsEnum(static_cast<T>(static_cast<Int>(this->value_) |
+                                        static_cast<Int>(flag)));
     }
 
-    FlagsEnum operator|(T flag)
+    constexpr FlagsEnum operator|(FlagsEnum rhs) const noexcept
     {
-        FlagsEnum xd;
-        xd.value_ = this->value_;
-        xd.set(flag, true);
+        return FlagsEnum(static_cast<T>(static_cast<Int>(this->value_) |
+                                        static_cast<Int>(rhs.value_)));
+    }
+
+    constexpr bool has(T flag) const noexcept
+    {
+        return static_cast<Int>(this->value_) & static_cast<Int>(flag);
+    }
 
-        return xd;
+    constexpr bool hasAny(FlagsEnum flags) const noexcept
```

Comment:
does `FlagsEnum` without a template parameter mean "FlagsEnum with the same template parameter"?

Reply:
Yes. Unfortunately, I don't know how that's called.